### PR TITLE
fix(shell): Allow a single detached shell-panel at the end in content-behind mode

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #4156

## Summary

fix(shell): Allow a single detached shell-panel at the end in content-behind mode (#4156)

### CODE REVIEW

File: src/components/shell/shell.tsx [hunk 1]
```diff
@@ -11,8 +11,10 @@ import {
  * @slot - A slot for adding content to the shell. This content will appear between any leading and trailing panels added to the shell. (eg. a map)
  * @slot header - A slot for adding header content. This content will be positioned at the top of the shell.
  * @slot footer - A slot for adding footer content. This content will be positioned at the bottom of the shell.
- * @slot primary-panel - A slot for adding the leading `calcite-shell-panel`.
- * @slot contextual-panel - A slot for adding the trailing `calcite-shell-panel`.
+ * @slot panel-start - A slot for adding the starting `calcite-shell-panel`.
+ * @slot panel-end - A slot for adding the ending `calcite-shell-panel`.
+ * @deprecated @slot primary-panel - A slot for adding the leading `calcite-shell-panel`.
```
Comment (driskull) [inline src/components/shell/shell.tsx:16]: Is this the correct way to deprecate a slot? cc @benelan

**Comment by driskull** [comment]
@asangma @macandcheese this is ready for review

**Comment by macandcheese** [comment]
Screen Shot 2022-07-22 at 10 01 25 AM

File: src/demos/shell/demo-app-center-row-action-bar.html [hunk 2]
```diff
@@ -53,11 +53,12 @@
       });
     </script>
   </head>
+
   <body>
     <main>
       <div class="shell-container">
         <calcite-shell>
-          <calcite-shell-panel id="primary-panel" slot="primary-panel" position="start" id="primary-panel">
```
Comment (macandcheese) [inline src/demos/shell/demo-app-center-row-action-bar.html:60]: Just demo stuff, but a duplicate `id` here.

**Review by macandcheese — APPROVED** [review]
Awesome! Naming is much clearer now. I suppose we could have the slots as `shell-panel-start/end` to be a bit more clear that users should slot `shell-panel`, not `panel`, but will defer to others on that.

File: src/components/shell/shell.tsx (hunk 1 above)
Comment (benelan) [reply src/components/shell/shell.tsx:16]: cc @macandcheese @geospatialem you both found out the jsdoc tag doesn't work for slots as well as css vars right? IIRC we use `[Deprecated]` and then parse it out and add the chip in the doc.

> @ slot primary-panel - [Deprecated] A slot for adding the leading `calcite-shell-panel`.

(added a space between `@` and `slot` so I didn't tag some random guy)

**Comment by driskull** [comment]
> I suppose we could have the slots as shell-panel-start/end to be a bit more clear that users should slot shell-panel, not panel, but will defer to others on that.

@jcfranco what do you think about the slot names?

File: src/components/shell/shell.tsx (hunk 1 above)
Comment (benelan) [reply src/components/shell/shell.tsx:16]: @driskull I confirmed that adding `[DEPRECATED]` to the description is the correct way to deprecate slots and css vars. We will get that added to the documentation conventions.

**Comment by driskull** [comment]
I changed the slot names to `shell-panel-start` and `shell-panel-end`. @jcfranco let me know if that should change to something else. Otherwise, I think this is good to merge.

**Comment by jcfranco** [comment]
I think the original names (sans `shell`) are a good balance between concise and meaningful. Most of our slot names are contextual and aren't 100% tied to the exact elements that get slotted.

**Comment by driskull** [comment]
@jcfranco I'll put them back to as they were.

Maybe we should discuss the naming of slots more. Some slots like `actions-start` do suggest some type of content that should be slotted.

In this case, it might help distinguish the `calcite-panel` from the `calcite-shell-panel`.

**Comment by jcfranco** [comment]
Yeah, I can see that, but for slots, users are sort of forced to go through the doc/examples as slots don't come through the type declarations nor can be easily discovered like props /methods in the console. I would lean more on adding `shell` this wasn't the case. Do you recall any confusion w/ `primary/secondary-panel`?

**Review by jcfranco — APPROVED** [review]
:boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom:
:boom::bomb::bomb::bomb::bomb::boom::boom::bomb::bomb::boom::boom::boom::bomb::bomb::boom::boom::bomb::boom::boom::boom::bomb::boom::bomb::boom:
:boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::bomb::boom::bomb::bomb::boom::bomb::boom:
:boom::bomb::bomb::bomb::boom::boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::boom::bomb::boom::bomb::boom::bomb::boom:
:boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::boom::bomb::boom::boom::boom:
:boom::bomb::bomb::bomb::bomb::boom::boom::bomb::bomb::boom::boom::boom::bomb::bomb::boom::boom::bomb::boom::boom::boom::bomb::boom::bomb::boom:
:boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom:

**Pull request merged**